lib/libevent.c

```diff
@@ -0,0 +1,246 @@
+/*
+ * libwebsockets - small server side websockets and web server implementation
+ *
+ * Copyright (C) 2010-2014 Andy Green <andy@warmcat.com>
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation:
+ *  version 2.1 of the License.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ *  MA  02110-1301  USA
+ */
+
+#include "private-libwebsockets.h"
+
+void lws_feature_status_libevent(struct lws_context_creation_info *info)
+{
+  if (lws_check_opt(info->options, LWS_SERVER_OPTION_LIBEVENT))
+    lwsl_notice("libevent support compiled in and enabled\n");
+  else
+    lwsl_notice("libevent support compiled in but disabled\n");
+}
+
+static void
+lws_event_cb(evutil_socket_t sock_fd, short revents, void *ctx)
+{
+  struct lws_io_watcher *lws_io = (struct lws_io_watcher *)ctx;
+  struct lws_context *context = lws_io->context;
+  struct lws_pollfd eventfd;
+
+  if (revents & EV_TIMEOUT)
+    return;
+
+  if (revents & EV_CLOSED)
```

Comment:
Is it possible to have `EV_READ` or `EV_READ` in addition to `EV_CLOSE`?  I think it might in which case we should check those flags first prior to `EV_CLOSED` and `EV_TIMEOUT`.